### PR TITLE
Add sample_qs_{X}_to_{Y} field and update transcriptConsequenceTerms field.

### PR DESCRIPTION
**Pull request by ShifaSZ**
See details in issue: https://github.com/broadinstitute/seqr-private/issues/1019

### CODE REVIEW

File: sv_pipeline/genome/load_data.py [hunk 1]
```diff
@@ -42,19 +47,22 @@
...
-                                              hl.if_else((rows.sv_type[0] == 'INS') & (hl.len(rows.sv_type) > 1),
-                                                         rows.sv_type[1], hl.missing('str'))),
+    'svType': lambda rows: rows[SV_TYPE][0],
+    TRANS_CONSEQ_TERMS: lambda rows: [rows[SV_TYPE][0]],
+    'sv_type_detail': lambda rows: hl.if_else(rows[SV_TYPE][0] == 'CPX', rows.info.CPX_TYPE,
+                                              hl.if_else((rows[SV_TYPE][0] == 'INS') & (hl.len(rows[SV_TYPE]) > 1),
+                                                         rows[SV_TYPE][1], hl.missing('str'))),
     'geneIds': lambda rows: hl.set(hl.map(lambda x: x.gene_id, rows.sortedTranscriptConsequences.filter(
-        lambda x: x.major_consequence != 'NEAREST_TSS'))),
+        lambda x: x[MAJOR_CONSEQ] != 'NEAREST_TSS'))),
```
Comment (ShifaSZ) [inline sv_pipeline/genome/load_data.py:56]: These are just changes for using constant field names.

File: sv_pipeline/genome/load_data.py [hunk 2]
```diff
@@ -129,20 +142,25 @@ def annotate_fields(mt, gencode_release, gencode_path):
...
+        SORTED_TRANS_CONSEQ: hl.flatmap(lambda x: x, hl.filter(
             lambda x: hl.is_defined(x),
-            [rows.info[col].map(lambda gene: hl.struct(gene_symbol=gene, gene_id=gene_id_mapping[gene],
-                                                       major_consequence=col.split('__')[-1]))
+            [rows.info[col].map(lambda gene: hl.struct(**{'gene_symbol': gene, 'gene_id': gene_id_mapping[gene],
+                                                       MAJOR_CONSEQ: col.split('__')[-1]}))
              for col in [gene_col for gene_col in rows.info if gene_col.startswith('PROTEIN_CODING__')
                          and rows.info[gene_col].dtype == hl.dtype('array<str>')]])),
-        sv_type=rows.alleles[1].replace('[<>]', '').split(':', 2),
+        SV_TYPE: rows.alleles[1].replace('[<>]', '').split(':', 2)}
```
Comment (ShifaSZ) [inline sv_pipeline/genome/load_data.py:152]: Just change for using constant field names `SORTED_TRANS_CONSEQ`, `MAJOR_CONSEQ`, and `SV_TYPE`.

**Comment by ShifaSZ** [comment]
An example variant imported to the Elasticsearch:
```
{
  "_index": "r0486_cmg_gcnv__structural_variants__wgs__grch38__20210720",
  "_type": "_doc",
  "_id": "g51lxXoBi-gUVj-0g-5l",
  "_score": 1.0,
  "_source": {
    "contig": "8",
    "sc": 23,
    "sf": 0.016106,
    "sn": 1428,
    "start": 77815735,
    "end": 77816226,
    "sv_callset_Het": 23,
    "sv_callset_Hom": 0,
    "pos": 77815735,
    "filters": [
      "FAIL_minGQ"
    ],
    "xpos": 8077815735,
    "xstart": 8077815735,
    "xstop": 8077816226,
    "svType": "DEL",
    "transcriptConsequenceTerms": [
      "DEL",
      "NEAREST_TSS"
    ],
    "geneIds": [],
    "samples_no_call": [],
    "samples_num_alt_1": [
      "GLE_4292_4_3",
      "GLE-4778-4-2-a",
      "HIL_A5192_12_1",
      "GLE_4292_4_1"
    ],
    "samples_num_alt_2": [],
    "variantId": "CMG.phase1_CMG_DEL_chr8_1792",
    "sortedTranscriptConsequences": [
      {
        "gene_symbol": "PKIA",
        "gene_id": "ENSG00000171033",
        "major_consequence": "NEAREST_TSS"
      }
    ],
    "genotypes": [
      {
        "sample_id": "GLE_4292_4_2",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE_4292_4_3",
        "gq": 1,
        "cn": 2,
        "num_alt": 1
      },
      {
        "sample_id": "GLE-4778-4-2-a",
        "gq": 1,
        "cn": 2,
        "num_alt": 1
      },
      {
        "sample_id": "GLE_4905_3_2",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "MCOP067",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "MWAR053",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE-3677-4-2-a",
        "gq": 5,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE-3677-4-4-a",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE-3871-4-2-a",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE_4397_4_2",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE_4397_4_3",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE-5279-3-3",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE_5483_3_3",
        "gq": 999,
        "cn": 1,
        "num_alt": 0
      },
      {
        "sample_id": "HIL_A5192_12_1",
        "gq": 1,
        "cn": 2,
        "num_alt": 1
      },
      {
        "sample_id": "MBEL001",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "MPAU036",
        "gq": 999,
        "cn": 1,
        "num_alt": 0
      },
      {
        "sample_id": "VS_ABC004_1",
        "gq": 999,
        "cn": 1,
        "num_alt": 0
      },
      {
        "sample_id": "VS_EU003_1",
        "gq": 70,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE-3386-4-4-a",
        "gq": 85,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE-3386-4-6-a",
        "gq": 999,
        "cn": 1,
        "num_alt": 0
      },
      {
        "sample_id": "GLE-3414-4-3-a",
        "gq": 999,
        "cn": 1,
        "num_alt": 0
      },
      {
        "sample_id": "GLE-3414-4-6-a",
        "gq": 120,
        "cn": 1,
        "num_alt": 0
      },
      {
        "sample_id": "GLE-3734-4-4-a",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE-3822-4-3-a",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE_3822_4_4",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE_4111_4_5",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "GLE_4292_4_1",
        "gq": 1,
        "cn": 2,
        "num_alt": 1
      },
      {
        "sample_id": "GLE-5279-3-1",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "HIL_A5192_21_1",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "MBEL122",
        "gq": 999,
        "cn": 1,
        "num_alt": 0
      },
      {
        "sample_id": "MCAI002",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "MCAI043",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "MCAI054",
        "gq": 999,
        "cn": 1,
        "num_alt": 0
      },
      {
        "sample_id": "MWAR044",
        "gq": 999,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "VS_EU003_2",
        "gq": 120,
        "cn": 2,
        "num_alt": 0
      },
      {
        "sample_id": "VS_EU003_3",
        "gq": 79,
        "cn": 2,
        "num_alt": 0
      }
    ],
    "sample_qs_0_to_10": [
      "GLE_4292_4_3",
      "GLE-4778-4-2-a",
      "GLE-3677-4-2-a",
      "HIL_A5192_12_1",
      "GLE_4292_4_1"
    ],
    "sample_qs_70_to_80": [
      "VS_EU003_1",
      "VS_EU003_3"
    ],
    "sample_qs_80_to_90": [
      "GLE-3386-4-4-a"
    ],
    "sample_qs_120_to_130": [
      "GLE-3414-4-6-a",
      "VS_EU003_2"
    ],
    "sample_qs_990_to_1000": [
      "GLE_4292_4_2",
      "GLE_4905_3_2",
      "MCOP067",
      "MWAR053",
      "GLE-3677-4-4-a",
      "GLE-3871-4-2-a",
      "GLE_4397_4_2",
      "GLE_4397_4_3",
      "GLE-5279-3-3",
      "GLE_5483_3_3",
      "MBEL001",
      "MPAU036",
      "VS_ABC004_1",
      "GLE-3386-4-6-a",
      "GLE-3414-4-3-a",
      "GLE-3734-4-4-a",
      "GLE-3822-4-3-a",
      "GLE_3822_4_4",
      "GLE_4111_4_5",
      "GLE-5279-3-1",
      "HIL_A5192_21_1",
      "MBEL122",
      "MCAI002",
      "MCAI043",
      "MCAI054",
      "MWAR044"
    ]
  }
}
```

File: sv_pipeline/genome/load_data.py [hunk 3]
```diff
@@ -42,19 +47,22 @@
...
+        lambda x: x[MAJOR_CONSEQ] != 'NEAREST_TSS'))),
     'samples_no_call': lambda rows: get_sample_num_alt_x(rows, -1),
     'samples_num_alt_1': lambda rows: get_sample_num_alt_x(rows, 1),
     'samples_num_alt_2': lambda rows: get_sample_num_alt_x(rows, 2),
 }
 
-FIELDS = list(CORE_FIELDS.keys()) + list(DERIVED_FIELDS.keys()) + ['variantId', 'sortedTranscriptConsequences', 'genotypes']
+SAMPLE_QS_FIELDS = {i: 'sample_qs_{}_to_{}'.format(i, i+QS_BIN_SIZE) for i in range(0, 1000, QS_BIN_SIZE)}
+
+FIELDS = list(CORE_FIELDS.keys()) + list(DERIVED_FIELDS.keys()) + ['variantId', 'sortedTranscriptConsequences', 'genotypes'] +\
```
Comment (hanars) [inline sv_pipeline/genome/load_data.py:64]: you could use the `SORTED_TRANS_CONSEQ` constant here as well

File: sv_pipeline/genome/load_data.py [hunk 4]
```diff
@@ -129,20 +142,25 @@ def annotate_fields(mt, gencode_release, gencode_path):
...
+        SV_TYPE: rows.alleles[1].replace('[<>]', '').split(':', 2)}
     )
 
     DERIVED_FIELDS.update({'filters': lambda rows: hl.if_else(hl.len(rows.filters) > 0, rows.filters,
                                                                  hl.missing(hl.dtype('array<str>')))})
     rows = rows.annotate(**{k: v(rows) for k, v in DERIVED_FIELDS.items()})
 
+    rows = rows.annotate(**{v: get_sample_in_gq_range(rows, i, i+QS_BIN_SIZE) for i, v in SAMPLE_QS_FIELDS.items()})
+
+    rows = rows.annotate(**{TRANS_CONSEQ_TERMS: rows[TRANS_CONSEQ_TERMS]
```
Comment (hanars) [inline sv_pipeline/genome/load_data.py:161]: It is much cleaner to just modify the function for this field in `DERIVED_FIELDS`

**Review by hanars — CHANGES_REQUESTED** [review]

**Review by hanars — APPROVED** [review]
looks good to me!

**Pull request merged**